docs/UPLOAD-UTILITY.md

```diff
@@ -0,0 +1,93 @@
+# Uploading Previously Collected Test Data to an Instance of Murakami-Viz
+
+[Murakami-Viz](https://github.com/m-lab/murakami-viz/) is a separate data visualization service that can optionally be used with Murakami measurement devices, particularly a fleet of devices managed through Balena.io.
+
+Included with Murakami is a utility script, **murakami-upload** found in `scripts/murakami-upload.py`, that provides a means of uploading previously collected test results from Murakami's NDT or Speedtest-CLI runners to an instance of Murakami-Viz.
+
+For `murakami-upload` to successfully upload results to a Murakami Viz instance:
+* all test result files must contain the MurakamiDeviceID field
+* each device ID must be associated with a measurement device in the Murakami-Viz instance
+* the current, public IP address of the computer you’ll use to run murakami-upload must be whitelisted in the Murakami-Viz instance
+
+## How to Use murakami-upload
+
+* Obtain test data from missing dates
+* Enter or Start an interactive Murakami container containing the missing test data
+* Run murakami-upload
+* Confirm data was received by the Murakami-Viz instance
+
+### Obtain test data from on-device or exported locations
+
+Depending on the export locations configured for your Murakami measurement device, test results may be stored in one or more locations.
+
+#### On-Device Data
+
+To upload test data that is stored on a Murakami measurement device, first login to the device itself. If this is a Balena Cloud managed device, you can use the terminal through your Balena app or the balena-cli tool to login to the main container. If this is a standalone device, SSH to the device.
+
+In either case, when you’re on the device’s terminal, first confirm that murakami-upload is a usable program in the Murakami release running on the device.
+
+```
+root@d9fa4d8:/murakami# ls scripts/
```

Comment:
Perhaps just `which murakami-upload` would be enough?

Reply:
Sure, though I wanted to show the system path in case people want to use it with Python directly.